modules/issue_tracker/php/endpoints/attachment.class.inc

```diff
@@ -0,0 +1,245 @@
+            $response
+        );
+    }
+
+    /**
+     * Handle DELETE requests for attachment
+     * and marks attachment as deleted from issue_tracker.
+     *
+     * @param ServerRequestInterface $request The incoming PSR7 request.
+     *
+     * @return ResponseInterface The outgoing PSR7 response
+     */
+    private function _handleDELETE(ServerRequestInterface $request)
+    : ResponseInterface
+    {
+        // Verify User can delete file.
+        if ($this->_deletePermission($request)) {
+            // Parse request body.
+            $values    = $request->getQueryParams();
+            $file_hash = $values['file_hash'];
+            $factory   = \NDB_Factory::singleton();
+            $DB        = $factory->database();
+            $DB->update(
+                'issues_attachments',
+                array('deleted' => 1),
+                array('file_hash' => $file_hash)
+            );
+            return new \LORIS\Http\Response\JsonResponse(
+                array('success' => true)
+            );
+        }
+        return new \LORIS\Http\Response\JSON\Forbidden(
+            'Invalid Permissions'
+        );
+    }
+
+    /**
+     * Check if user is able to delete the attachment
+     * from the issue_tracker module.
+     *
+     * @param ServerRequestInterface $request The incoming PSR7 request.
+     *
+     * @return bool The permission to delete attachment.
+     */
+    private function _deletePermission(ServerRequestInterface $request)
+    : bool
+    {
+        // Parse request body.
+        $values   = $request->getQueryParams();
+        $user     = \NDB_Factory::singleton()->user();
+        $username = $user->getUsername();
+        $ID       = $values['ID'];
+        $DB       = \NDB_Factory::singleton()->database();
+        $query    = 'SELECT
+                        USER
+                    FROM
+                        issues_attachments
+                    WHERE
+                        ID = :ID
+                        AND USER = :user;';
+        $result   = $DB->pselectOne(
+            $query,
+            array('ID' => $ID, 'user' => $username)
+        );
+        // Check if User shouldn't be able to Delete.
+        if (empty($result)
+            && !$user->hasPermission(
+                'issue_tracker_developer'
+            )
+        ) {
+            return false;
+        }
+        return true;
+    }
+
+    /**
+     * An ETagCalculator provides the ability to calculate an ETag for
+     * an incoming HTTP request.
+     *
+     * @param ServerRequestInterface $request The incoming PSR7 request.
+     *
+     * @return string The value to use for the ETag header.
+     */
+    public function ETag(ServerRequestInterface $request): string
+    {
+        if ($request->getMethod() === 'POST'
+            || $request->getMethod() === 'DELETE'
+            || $request->getMethod() === 'GET'
+        ) {
+            return 'Etag not calculated for POST, DELETE, and GET request';
```

Comment:
Why are "Etag not calculated for POST, DELETE, and GET request" ?
DELETE should not but GET request should have an ETag; specially if the attachment is big.
And sending an ETag after a post would be a nice curtesy.

Reply:
@xlecours I found out from @jesscall testing the PR that Etag caching breaks it.